File: php/SimplyPay/SimplyPayAPI.php
<?

	require_once ("HttpConnection.php");
	
	class SimplyPayAPI {
		
		
		public function __construct() 
	    { 
	        $a = func_get_args(); 
	        $i = func_num_args(); 
	        if (method_exists($this,$f='__construct'.$i)) { 
	            call_user_func_array(array($this,$f),$a); 
	        } 
	    } 
	    
		
		public function __construct2($realm , $rsaKey) {
			
			
			if (!$realm || $realm == "") {
				throw new Exception("REALM 값이 필요합니다");
			}
			
			if (!$rsaKey || $rsaKey == "") {
				throw new Exception("RSA Key값이 필요합니다");
			}
			
			
			$this->realm = $realm;
			$this->rsaKey = $rsaKey;
			
			$rsa = new Crypt_RSA();
			$rsa->setEncryptionMode(CRYPT_RSA_ENCRYPTION_PKCS1);
		    $rsa->loadKey($this->rsaKey); 
		    
		    $this->rsa = $rsa;
			
			
		}
		
		public function __construct4($realm , $rsaKey,$aesKey,$token) {
			$this->__construct2($realm,$rsaKey);
			$this->aesKey = $aesKey;
			$this->token = $token;
			
			try {
				base64_decode($aesKey);
			}catch(Exception $e) {
				throw new Exception("올바르게 암호화된 AES KEY가 아닙니다.");
			}
			
			if (!$token || $token == "") {
				throw new Exception("세션 토큰이 필요합니다.");
			}
			
		}
		

		private function aes_encrypt($sKey,$sStr) {
		  return base64_encode(
		    mcrypt_encrypt(
		        MCRYPT_RIJNDAEL_128, 
		        base64_decode($sKey),
		        $this->pkcs5_pad($sStr,16),
		        MCRYPT_MODE_ECB
		    )
		  );
		}
		
		private function aes_decrypt($sKey,$sStr) {
		  return mcrypt_decrypt(
		    MCRYPT_RIJNDAEL_128, 
		    base64_decode($sKey), 
		    base64_decode($sStr), 
		    MCRYPT_MODE_ECB
		  );
		}
		
		private function pkcs5_pad ($text, $blocksize = 16) { 
		  $pad = $blocksize - (strlen($text) % $blocksize); 
		  return $text . str_repeat(chr($pad), $pad); 
		}
		
		private $DEBUG = true;
		private $rsaKey ;
		private $aesKey ;
		private $realm;
		private $token ;
		private $host = "www.kkokjee.com" ;
		private $scheme = "https" ;
		private $port = 443;
		private $rsa;
		
		private $encTable = array(
			"token"=>"plain",
			"realm"=>"plain",
			"callback_url"=>"plain",
			"screen_name"=>"rsa"	
		);
		
		public static $CODE_OK = 200;
		public static $CODE_ERROR = 400;		
		
		private static $PATH_CREATE_SESSION = "/simple/v1/create_session";
		private static $PATH_REGISTER_CARD = "/simple/v1/register_card";
		private static $PATH_LIST_CARD = "/simple/v1/card_list";
		private static $PATH_UNREGISTER_CARD = "/simple/v1/unregister_card";
		private static $PATH_PAY_CARD = "/simple/v1/do_pay_card";
		private static $PATH_CANCEL_CARD = "/simple/v1/do_cancel_card";
		
		
		public function setDevServer() {
			$this->host = "dev.kkokjee.com";
			$this->scheme = "http";
			$this->port = 80;
		}
		
		public function setRealServer() {
			$this->host = "www.kkokjee.com";
			$this->scheme = "https";
			$this->port = 443;
		}
		
		public function getToken() {
			return $this->token;
		}
		
		public function getAesKey() {
			return $this->aesKey;
		}
		
		public function encryptPostData($data) {
			$encdata = array();
			foreach($data as $key=>$value) {
				if (!isset($this->encTable[$key])) {
					if ($this->aesKey != null) {
						$encdata[$key] = $this->aes_encrypt($this->aesKey,$value);
					}
				}else {
					$enc = $this->encTable[$key];
					if ($enc == "plain") {
						$encdata[$key] = $value;
					}else {
						$encdata[$key] = base64_encode($this->rsa->encrypt($value));
					}
				}
			}
			return $encdata;
		}
		
		public function callApi($endpoint , $post_data) {
			
			if ($this->DEBUG) {
				echo "BEFORE ENCRYPTING\n";
				print_r($post_data);
			}


			$enc_data = $this->encryptPostData($post_data);
			$scheme = $this->scheme;
			$host = $this->host;
			$url = "${scheme}://${host}${endpoint}";
			
			if ($this->DEBUG) {
				echo "REQUEST : ${url}\n";
				print_r($enc_data);
			}
			
			$http = new HttpConnection("post",$url , $enc_data); 
		  	$json = json_decode($http->request());
			if ($this->DEBUG) {
				print_r($json);
			}
			return $json;
		}
		
		public function createSession($screen_name) {
			
			if (!$this->realm || $this->realm == "") {
				throw new Exception("REALM 값이 필요합니다");
			}
			
			if (!$screen_name || $screen_name == "") {
				throw new Exception("유저 아이디 값이 필요합니다");
			}
			
			$post_data = array(
				"realm"=>$this->realm,
				"screen_name"=>$screen_name,
				"callback_url"=>"reserved_for_future"
			);
			
			$json = $this->callApi(SimplyPayAPI::$PATH_CREATE_SESSION , $post_data);
			return $json;
		}
		
		
		public function registerCard(
			$screen_name,
			$title,
			$card_no,
			$card_date,
			$card_secure,
			$ssn,
			$pin
			
		) {
			
			try {
				base64_decode($this->aesKey);
			}catch(Exception $e) {
				throw new Exception("올바르게 암호화된 AES KEY가 아닙니다.");
			}
			
			if (!$this->token || $this->token == "") {
				throw new Exception("세션 토큰이 필요합니다.");
			}
			
			$post_data = array(
				"realm"=>$this->realm,
				"token"=>$this->token,
				"screen_name"=>$screen_name,
				"card_no"=>$card_no,
				"card_date"=>$card_date,
				"card_secure"=>$card_secure,
				"ssn"=>$ssn,
				"pin"=>$pin
			);
			
			$json = $this->callApi(SimplyPayAPI::$PATH_REGISTER_CARD , $post_data);
			return $json;
			
		}
		
		
		public function listCard($screen_name) {
			
			try {
				base64_decode($this->aesKey);
			}catch(Exception $e) {
				throw new Exception("올바르게 암호화된 AES KEY가 아닙니다.");
			}
			
			if (!$this->token || $this->token == "") {
				throw new Exception("세션 토큰이 필요합니다.");
			}
			
			$post_data = array(
				"realm"=>$this->realm,
				"token"=>$this->token,
				"screen_name"=>$screen_name
			);
			
			$json = $this->callApi(SimplyPayAPI::$PATH_LIST_CARD , $post_data);
			return $json;
		}
		
		
		public function payCard(
			$screen_name,
			$card_uuid,
			$pin,
			$installment,
			$amount,
			$tax,
			$fee,
			$comment = "",
			$additional_data = "",
			$customer_name = "",
			$customer_email = "",
			$customer_phone = "",
			$customer_mobile = ""
		) {
			
			try {
				base64_decode($this->aesKey);
			}catch(Exception $e) {
				throw new Exception("올바르게 암호화된 AES KEY가 아닙니다.");
			}
			
			if (!$this->token || $this->token == "") {
				throw new Exception("세션 토큰이 필요합니다.");
			}
			
			$post_data = array(
				"realm"=>$this->realm,
				"token"=>$this->token,
				"screen_name"=>$screen_name,
				"card_uuid"=>$card_uuid,
				"pin"=>$pin,
				"installment"=>$installment,
				"amount"=>$amount,
				"tax"=>$tax,
				"fee"=>$fee,
				
				"comment"=>$comment,
				"additional_data"=>$additional_data,
				
				"customer_name"=>$customer_name,
				"customer_email"=>$customer_email,
				"customer_phone"=>$customer_phone,
				"customer_mobile"=>$customer_mobile
			);
			
			$json = $this->callApi(SimplyPayAPI::$PATH_PAY_CARD , $post_data);
			return $json;
		}
		
		
		public function cancelCard(
			$screen_name,
			$order_no
		)
		{
			
			try {
				base64_decode($this->aesKey);
			}catch(Exception $e) {
				throw new Exception("올바르게 암호화된 AES KEY가 아닙니다.");
			}
			
			if (!$this->token || $this->token == "") {
				throw new Exception("세션 토큰이 필요합니다.");
			}
			
			$post_data = array(
				"realm"=>$this->realm,
				"token"=>$this->token,
				"screen_name"=>$screen_name,
				"order_no"=>$order_no
			);
			
			$json = $this->callApi(SimplyPayAPI::$PATH_CANCEL_CARD , $post_data);
			return $json;
		}
		
		
		public function unregisterCard(
			$screen_name,
			$card_uuid,
			$pin
		) {
			
			try {
				base64_decode($this->aesKey);
			}catch(Exception $e) {
				throw new Exception("올바르게 암호화된 AES KEY가 아닙니다.");
			}
			
			if (!$this->token || $this->token == "") {
				throw new Exception("세션 토큰이 필요합니다.");
			}
			
			$post_data = array(
				"realm"=>$this->realm,
				"token"=>$this->token,
				"screen_name"=>$screen_name,
				"card_uuid"=>$card_uuid,
				"pin"=>$pin
			);
			
			$json = $this->callApi(SimplyPayAPI::$PATH_UNREGISTER_CARD , $post_data);
			return $json;
		}
		
		
	}


?>

File: java/SimplyPay/src/simplypay/SimplyPayAPI.java
package simplypay;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.apache.commons.codec.binary.Base64;
import org.codehaus.jackson.JsonNode;


@SuppressWarnings("unused")
public class SimplyPayAPI {
	
	
	public boolean DEBUG = true;
	
	
	private String rsaKey = null;
	private String aesKey = null;
	private String realm = null;
	private String token = null;
	
	private String host = "www.kkokjee.com";
	private String scheme = "https";
	private int port = 443;
	
	
	private static final int CODE_OK = 200;
	private static final int CODE_ERROR = 400;
	
	private static final String PATH_CREATE_SESSION = "/simple/v1/create_session";
	private static final String PATH_REGISTER_CARD = "/simple/v1/register_card";
	private static final String PATH_LIST_CARD = "/simple/v1/card_list";
	private static final String PATH_UNREGISTER_CARD = "/simple/v1/unregister_card";
	private static final String PATH_PAY_CARD = "/simple/v1/do_pay_card";
	private static final String PATH_CANCEL_CARD = "/simple/v1/do_cancel_card";
	
	public void setDevServer() {
		host = "dev.kkokjee.com";
		scheme = "http";
		port = 80;
	}
	
	public void setRealServer() {
		host = "www.kkokjee.com";
		scheme = "https";
		port = 443;
	}
	
	public SimplyPayAPI(String realm , String rsaKey) throws SimplyPayException {
		
		this(realm,rsaKey,null,null);
	}
	
	public SimplyPayAPI(String realm,String rsaKey,String aesKey,String token) throws SimplyPayException{
		if (rsaKey == null || rsaKey.length() < 1) {
			throw new SimplyPayException("RSA 키가 필요합니다");
		}
		
		if (realm == null || realm.length() < 1) {
			throw new SimplyPayException("REALM 값이 필요합니다");
		}
		
		try {
			Base64.decodeBase64(rsaKey);
		}catch(Exception e) {
			throw new SimplyPayException("올바르게 인코딩된 RSA 키가 아닙니다");
		}
		
		this.realm = realm;
		this.rsaKey = rsaKey;
		
		if (aesKey != null && aesKey.length() > 1) {
			try {
				Base64.decodeBase64(aesKey);
			}catch(Exception e) {
				throw new SimplyPayException("올바르게 인코딩된 AES 키가 아닙니다");
			}
			
			this.aesKey = aesKey;
			
			if (token == null || token.length() < 1) {
				throw new SimplyPayException("세션 토큰값이 필요합니다");
			}
			
			this.token = token;
		}
	}
	
	public String getToken() {
		return token;
	}
	
	public String getAesKey() {
		return aesKey;
	}
	
	public void d(String msg) {
		if (DEBUG) {
			System.out.println("DEBUG ==> "+msg);
		}
	}

	public void createSession(String screen_name) throws SimplyPayException {
		
		if (realm == null || realm.length() < 1)  
			throw new SimplyPayException("REALM 값이 필요합니다");
		
		if (screen_name == null || screen_name.length() < 1) 
			throw new SimplyPayException("유저 아이디가 필요합니다");
		
		
		HashMap<String,Object> params = new HashMap<String,Object>();
		
		params.put("realm", realm);
		params.put("screen_name" , screen_name);
		params.put("callback_url" , "reserved_for_future");
		
		d(params.toString());
		
		params = SimplyPayUtil.encryptParams(rsaKey, null, params);
		
		d(params.toString());
		
		JsonNode json = SimplyPayUtil.callJsonNode(scheme,host,port,PATH_CREATE_SESSION,params);
		d("createSession : "+json.toString());
		
		try {
			if (json.get("code").asInt() != CODE_OK) {
				String message = json.get("message").asText();
				throw new SimplyPayException(message);
			}
			
			Map<String,Object> data = (Map<String,Object>) StringUtil.jsonToMap(json.get("data"));
			
			String aesKey = data.get("key").toString();
			String token = data.get("token").toString();

			
			if (aesKey != null && aesKey.length() > 1) {
				try {
					Base64.decodeBase64(aesKey);
				}catch(Exception e) {
					throw new SimplyPayException("올바르게 인코딩된 AES 키가 아닙니다");
				}
				
				this.aesKey = aesKey;
				
				if (token == null || token.length() < 1) {
					throw new SimplyPayException("올바른 토큰 값이 아닙니다");
				}
				
				this.token = token;
			}
			
		}
		catch(Exception e) {
			e.printStackTrace();
			throw new SimplyPayException(e);
		}
	}
	
	
	public boolean registerCard(
			String screen_name,
			String title,
			String card_no,
			String card_date,
			String card_secure,
			String ssn,
			String pin
		) throws SimplyPayException {
		
		
		if (realm == null || realm.length() < 1)  
			throw new SimplyPayException("REALM 값이 필요합니다");
		
		if (token == null || token.length() < 1) 
			throw new SimplyPayException("createSession이후 token값이 필요합니다");
		
		if (aesKey == null || aesKey.length() < 1) 
			throw new SimplyPayException("createSession이후 AES 키값이 필요합니다");
		
		if (screen_name == null || screen_name.length() < 1) 
			throw new SimplyPayException("유저 아이디가 필요합니다");
		
		if (title == null) title = "";
		
		if (card_no == null || card_no.length() < 14 ) 
			throw new SimplyPayException("카드번호는 최소 15자리입니다");
		
		if (card_date == null || card_date.length() < 6) 
			throw new SimplyPayException("YYYYMM형식의 카드 유효기간이 필요합니다");
		
		if (card_secure == null || card_secure.length() < 2) 
			throw new SimplyPayException("카드 비밀번호 앞 2자리가 필요합니다");
		
		if (ssn == null || ssn.length() < 6) 
			throw new SimplyPayException("개인의 경우 생년월일 6자리 , 사업자의 경우 사업자번호 10자리가 필요합니다");
		
		if (pin == null || pin.length() < 4) 
			throw new SimplyPayException("간편결제 비밀번호는 최소 4자리입니다");
		
		HashMap<String,Object> params = new HashMap<String,Object>();
		
		params.put("realm", realm);
		params.put("token",this.token);
		params.put("screen_name" , screen_name);
		params.put("title" , title);
		
		params.put("card_no" , card_no);
		params.put("card_date" , card_date);
		params.put("card_secure" , card_secure);
		
		params.put("ssn" , ssn);
		params.put("pin" , pin);
		
		d(params.toString());
		
		params = SimplyPayUtil.encryptParams(rsaKey, aesKey, params);
		
		d(params.toString());
		
		JsonNode json = SimplyPayUtil.callJsonNode(scheme,host,port,PATH_REGISTER_CARD,params);
		d("registerCard : "+json.toString());
		
		boolean ret = false;
		try {
			if (json.get("code").asInt() != CODE_OK) {
				String message = json.get("message").asText();
				throw new SimplyPayException(message);
			}else {
				ret = true;
			}
			
			Map<String,Object> data = (Map<String,Object>) StringUtil.jsonToMap(json.get("data"));
			
		}
		catch(Exception e) {
			e.printStackTrace();
			throw new SimplyPayException(e);
		}
		
		
		return ret;
	}
	
	
	public List<Map<?,?>> listCard(
			String screen_name
		) throws SimplyPayException {
		
		
		if (realm == null || realm.length() < 1)  
			throw new SimplyPayException("REALM 값이 필요합니다");
		
		if (token == null || token.length() < 1) 
			throw new SimplyPayException("createSession이후 token값이 필요합니다");
		
		if (aesKey == null || aesKey.length() < 1) 
			throw new SimplyPayException("createSession이후 AES 키값이 필요합니다");
		
		if (screen_name == null || screen_name.length() < 1) 
			throw new SimplyPayException("유저 아이디가 필요합니다");
		
		HashMap<String,Object> params = new HashMap<String,Object>();
		
		params.put("realm", realm);
		params.put("token",this.token);
		params.put("screen_name" , screen_name);
		
		d(params.toString());
		
		params = SimplyPayUtil.encryptParams(rsaKey, aesKey, params);
		
		d(params.toString());
		
		JsonNode json = SimplyPayUtil.callJsonNode(scheme,host,port,PATH_LIST_CARD,params);
		d("listCard : "+json.toString());
		
		List<Map<?,?>> list = null;
		
		try {
			if (json.get("code").asInt() != CODE_OK) {
				String message = json.get("message").asText();
				throw new SimplyPayException(message);
			}else {
				
			}
			
			Map<String,Object> data = (Map<String,Object>) StringUtil.jsonToMap(json.get("data"));
			list = ( List<Map<?,?>> ) data.get("list");
			
		}
		catch(Exception e) {
			e.printStackTrace();
			throw new SimplyPayException(e);
		}
		
		
		return list;
	}
	
	public Map<?,?> payCard(
			String screen_name,
			
			String card_uuid,
			String pin,
			
			int installment,
			int amount,
			int tax,
			int fee
	) throws SimplyPayException {
		return payCard(screen_name,card_uuid,pin,installment,amount,tax,fee,null,null,null,null,null,null);
	}
	
	public Map<?,?> payCard(
			String screen_name,
			
			String card_uuid,
			String pin,
			
			int installment,
			int amount,
			int tax,
			int fee,
			
			String comment
	) throws SimplyPayException {
		return payCard(screen_name,card_uuid,pin,installment,amount,tax,fee,comment,null,null,null,null,null);
	}
	
	public Map<?,?> payCard(
			String screen_name,
			
			String card_uuid,
			String pin,
			
			int installment,
			int amount,
			int tax,
			int fee,
			
			String comment,
			String additional_data
	) throws SimplyPayException {
		return payCard(screen_name,card_uuid,pin,installment,amount,tax,fee,comment,additional_data,null,null,null,null);
	}
	
			
	public Map<?,?> payCard(
			String screen_name,
			
			String card_uuid,
			String pin,
			
			int installment,
			int amount,
			int tax,
			int fee,
			
			String comment,
			String additional_data,
			
			String customer_name,
			String customer_email,
			String customer_phone,
			String customer_mobile

		) throws SimplyPayException {
		
		
		if (realm == null || realm.length() < 1)  
			throw new SimplyPayException("REALM 값이 필요합니다");
		
		if (token == null || token.length() < 1) 
			throw new SimplyPayException("createSession이후 token값이 필요합니다");
		
		if (aesKey == null || aesKey.length() < 1) 
			throw new SimplyPayException("createSession이후 AES 키값이 필요합니다");
		
		if (screen_name == null || screen_name.length() < 1) 
			throw new SimplyPayException("유저 아이디가 필요합니다");
		
		if (card_uuid == null || card_uuid.length() < 32 ) 
			throw new SimplyPayException("카드 고유 번호가 필요합니다");
		
		if (pin == null || pin.length() < 4) 
			throw new SimplyPayException("간편결제 비밀번호는 최소 4자리입니다");
		
		
		if (comment == null || comment.length() < 1)  {
			comment = "";
		}
		
		if (additional_data == null || additional_data.length() < 1)  {
			additional_data = "";
		}
		
		if (customer_name == null || customer_name.length() < 1)  {
			customer_name = "";
		}
		if (customer_email == null || customer_email.length() < 1)  {
			customer_email = "";
		}
		if (customer_phone == null || customer_phone.length() < 1)  {
			customer_phone = "";
		}
		if (customer_mobile == null || customer_mobile.length() < 1)  {
			customer_mobile = "";
		}
		
		
		HashMap<String,Object> params = new HashMap<String,Object>();
		
		params.put("realm", realm);
		params.put("token",this.token);
		params.put("screen_name" , screen_name);
		
		params.put("card_uuid" , card_uuid);
		params.put("pin" , pin);
		
		params.put("installment" , installment+"");
		params.put("amount" , amount+"");
		params.put("tax" , tax+"");
		params.put("fee" , fee+"");
		
		
		params.put("comment" , comment);
		params.put("additional_data" , additional_data);
		
		params.put("customer_name" , customer_name);
		params.put("customer_email" , customer_email);
		params.put("customer_phone" , customer_phone);
		params.put("customer_mobile" , customer_mobile);
		
		
		
		d(params.toString());
		
		params = SimplyPayUtil.encryptParams(rsaKey, aesKey, params);
		
		d(params.toString());
		
		JsonNode json = SimplyPayUtil.callJsonNode(scheme,host,port,
				PATH_PAY_CARD,params);
		d("payCard : "+json.toString());
		
		Map<String,Object> data = null;
		
		try {
			if (json.get("code").asInt() != CODE_OK) {
				String message = json.get("message").asText();
				throw new SimplyPayException(message);
			}else {
				
			}
			
			data = (Map<String,Object>) StringUtil.jsonToMap(json.get("data"));
			
		}
		catch(Exception e) {
			e.printStackTrace();
			throw new SimplyPayException(e);
		}
		
		
		return data;
	}
	
	
	public Map<?,?> cancelCard(
			String screen_name,
			
			String order_no

		) throws SimplyPayException {
		
		
		if (realm == null || realm.length() < 1)  
			throw new SimplyPayException("REALM 값이 필요합니다");
		
		if (token == null || token.length() < 1) 
			throw new SimplyPayException("createSession이후 token값이 필요합니다");
		
		if (aesKey == null || aesKey.length() < 1) 
			throw new SimplyPayException("createSession이후 AES 키값이 필요합니다");
		
		if (screen_name == null || screen_name.length() < 1) 
			throw new SimplyPayException("유저 아이디가 필요합니다");
		
		if (order_no == null || order_no.length() < 1 ) 
			throw new SimplyPayException("거래 고유 주문번호가 필요합니다");

		
		HashMap<String,Object> params = new HashMap<String,Object>();
		
		params.put("realm", realm);
		params.put("token",this.token);
		params.put("screen_name" , screen_name);
		
		params.put("order_no" , order_no);
		
		
		d(params.toString());
		
		params = SimplyPayUtil.encryptParams(rsaKey, aesKey, params);
		
		d(params.toString());
		
		JsonNode json = SimplyPayUtil.callJsonNode(scheme,host,port,
				PATH_CANCEL_CARD,params);
		d("cancelCard : "+json.toString());
		
		Map<String,Object> data = null;
		
		try {
			if (json.get("code").asInt() != CODE_OK) {
				String message = json.get("message").asText();
				throw new SimplyPayException(message);
			}else {
				
			}
			
			data = (Map<String,Object>) StringUtil.jsonToMap(json.get("data"));
			
		}
		catch(Exception e) {
			e.printStackTrace();
			throw new SimplyPayException(e);
		}
		
		
		return data;
	}
	
	
	public boolean unregisterCard(
			String screen_name,
			
			String card_uuid,
			String pin
			

		) throws SimplyPayException {
		
		
		if (realm == null || realm.length() < 1)  
			throw new SimplyPayException("REALM 값이 필요합니다");
		
		if (token == null || token.length() < 1) 
			throw new SimplyPayException("createSession이후 token값이 필요합니다");
		
		if (aesKey == null || aesKey.length() < 1) 
			throw new SimplyPayException("createSession이후 AES 키값이 필요합니다");
		
		if (screen_name == null || screen_name.length() < 1) 
			throw new SimplyPayException("유저 아이디가 필요합니다");
		
		if (card_uuid == null || card_uuid.length() < 32 ) 
			throw new SimplyPayException("카드 고유 번호가 필요합니다");
		
		if (pin == null || pin.length() < 4) 
			throw new SimplyPayException("간편결제 비밀번호는 최소 4자리입니다");
		
		
		HashMap<String,Object> params = new HashMap<String,Object>();
		
		params.put("realm", realm);
		params.put("token",this.token);
		params.put("screen_name" , screen_name);
		
		params.put("card_uuid" , card_uuid);
		params.put("pin" , pin);
		
		
		d(params.toString());
		
		params = SimplyPayUtil.encryptParams(rsaKey, aesKey, params);
		
		d(params.toString());
		
		JsonNode json = SimplyPayUtil.callJsonNode(scheme,host,port,
				PATH_UNREGISTER_CARD,params);
		d("unregisterCard : "+json.toString());
		
		boolean ret = false;
		
		try {
			if (json.get("code").asInt() != CODE_OK) {
				String message = json.get("message").asText();
				throw new SimplyPayException(message);
			}else {
				ret = true;
			}
			
//			(Map<String,Object>) StringUtil.jsonToMap(json.get("data"));
			
		}
		catch(Exception e) {
			e.printStackTrace();
			throw new SimplyPayException(e);
		}
		
		
		return ret;
	}
	
}
